File: kalitech-engine/src/main/resources/kalitech/engine/modules/Mesh/Mesh.js
// FILE: resources/kalitech/engine/modules/Mesh/Mesh.js
// Author: Calista Verner
"use strict";

const META = Object.freeze({
    moduleId: "mesh",
    globalName: "MESH",
    version: "1.0.2",
    engineMin: "0.1.5",
    description: "RootKit wrapper over ENGINE.mesh(): create() returns object-mesh with physics methods via ENGINE.physics()"
});

const MeshOrchestrator = require("./MeshOrchestrator.js");

function create(ENGINE /*, K */) {
    if (!ENGINE) throw new Error("[MESH] ENGINE is required");
    const orch = new MeshOrchestrator(ENGINE);
    return orch.decorateMeshApi();
}

module.exports = create;
module.exports.META = META;

File: kalitech-engine/src/main/resources/kalitech/engine/modules/Mesh/Mesh.lua
local M = {}
local luaRuntime = require("@builtin/lua_runtime")
local Classes = luaRuntime.class
local Error = luaRuntime.Error
META = KObject:freeze({
    moduleId = "mesh",
    version = "1.0.2",
    engineMin = "0.1.5",
    description = "RootKit wrapper over ENGINE.mesh(): create() returns object-mesh with physics methods via ENGINE.physics()"
})
MeshOrchestrator = require("./MeshOrchestrator.lua")
function create(self, ENGINE)
    if not ENGINE then
        error(
            Classes:construct(Error, "[MESH] ENGINE is required"),
            0
        )
    end
    local orch = Classes:construct(MeshOrchestrator, ENGINE)
    return orch:decorateMeshApi()
end
M = setmetatable({META = META}, {
    __call = function(_, ...)
        return create(...)
    end
})

return M
